src/cfn-mcp-server/awslabs/cfn_mcp_server/impl/tools/resource_scanner_tools.py

```diff
@@ -0,0 +1,267 @@
+            + f'\n\nExample usage:\n'
+            f'- Single resource type: ["AWS::S3::Bucket"]\n'
+            f'- Multiple types: {common_resource_types}\n'
+            f'- Entire account: []'
+        )
+
+    try:
+        cfn_utils = CloudFormationUtils(region=region or 'us-east-1')
+    except Exception as e:
+        raise handle_aws_api_error(e)
+
+    try:
+        scan_id = cfn_utils.start_resource_scan(resource_types)
+        return {
+            'scan_id': scan_id,
+        }
+    except Exception as e:
+        raise handle_aws_api_error(e)
+
+
+async def list_resources_by_filter_impl(
+    resource_identifier: Optional[str] = None,
+    resource_type_prefix: Optional[str] = None,
+    tag_key: Optional[str] = None,
+    tag_value: Optional[str] = None,
+    limit: int = 100,
+    next_token: Optional[str] = None,
+    region: Optional[str] = None,
+) -> Dict[str, Any]:
+    """Simple implementation for listing AWS resources with filtering.
+
+    Args:
+        resource_identifier: Filter by specific resource identifier (e.g., physical ID)
+        resource_type_prefix: Filter by resource type prefix (e.g., "AWS::S3::")
+        tag_key: Filter resources by tag key
+        tag_value: Filter resources by tag value (requires tag_key)
+        limit: Maximum number of resources to return (1-100)
+        next_token: AWS pagination token from previous response
+        region: AWS region to use
+
+    Returns:
+        Dict containing filtered resources and pagination info
+    """
+    # Basic validation
+    if limit < 1 or limit > 100:
+        raise ClientError('Limit must be between 1 and 100')
+
+    try:
+        cfn_utils = CloudFormationUtils(region=region or 'us-east-1')
+
+        # Get latest completed scan
+        scans = cfn_utils.list_resource_scans()
+        latest_scan = next((s for s in scans if s.get('Status') == 'COMPLETE'), None)
+        if not scans:
+            raise ClientError('No resource scans found in account. Please start a resource scan.')
+        if not latest_scan:
+            raise ClientError(
+                'No completed resource scans found. Scan could still be in progress or start a resource scan if you none exists.'
+            )
+
+        # Get filtered resources from AWS API
+        if resource_type_prefix or tag_key or tag_value or resource_identifier:
+            result = cfn_utils.list_resource_scan_resources_with_filters(
+                scan_id=latest_scan['ResourceScanId'],
+                resource_identifier=resource_identifier,
+                resource_type_prefix=resource_type_prefix,
+                tag_key=tag_key,
+                tag_value=tag_value,
+                max_results=limit,
+                next_token=next_token,
+            )
+            resources = result['resources']
+        else:
+            # Get all resources if no filters
+            all_resources = cfn_utils.list_resource_scan_resources(latest_scan['ResourceScanId'])
```

Comment:
shouldn't `next_token` be applied to `all_resources`? Maybe just use the `list_resource_scan_resources_with_filters function`?